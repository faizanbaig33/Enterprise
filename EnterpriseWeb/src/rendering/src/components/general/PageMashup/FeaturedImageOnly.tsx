import { Field } from '@sitecore-jss/sitecore-jss-nextjs';
import classNames from 'classnames';
import { useTheme } from 'lib/context/ThemeContext';
import { getEnum } from 'lib/utils';
import { Eyebrow } from 'src/helpers/Eyebrow';
import { Headline } from 'src/helpers/Headline';
import { ImagePrimary } from 'src/helpers/Media';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import { ResultItem } from './ImagesForAll';
import { MashupStyle, PageMashupProps } from './PageMashup';
import { PageMashupTheme } from './PageMashup.theme';
import { RichTextWrapper } from 'src/helpers/RichTextWrapper';

type FeaturedImageOnlyProps = PageMashupProps & {
  fields: {
    resultItems: Array<ResultItem>;
  };
} & { mashupStyle: MashupStyle };
const FeaturedImageOnly = (props: FeaturedImageOnlyProps) => {
  const { themeData } = useTheme(PageMashupTheme);
  const ARTICLE_TEMPLATE_ID = '9256edf3-d0fa-4588-adda-3036b9d04faa';
  const fallBackImage = props.fields.fallbackImage;
  const featuredArticleDesktopImage = props.fields.resultItems[0].fields.articleImage.value?.src
    ? props.fields.resultItems[0].fields.articleImage
    : fallBackImage;

  const featuredPrimaryImage = props.fields.resultItems[0].fields.primaryImage.value?.src
    ? props.fields.resultItems[0].fields.primaryImage
    : fallBackImage;

  const featuredArticleMobileImage = props.fields.resultItems[0].fields.articleThumbnail?.value?.src
    ? props.fields.resultItems[0].fields.articleThumbnail
    : featuredArticleDesktopImage;

  const featuredPrimaryImageMobile = props.fields.resultItems[0].fields.primaryImageMobile?.value
    ?.src
    ? props.fields.resultItems[0].fields.primaryImageMobile
    : featuredPrimaryImage;

  return (
    <>
      <div
        className={classNames(
          props.mashupStyle != 'feature-image-only'
            ? themeData.classes.featuredCard.wrapper.noImages
            : themeData.classes.featuredCard.wrapper.featuredImage
        )}
      >
        {props.mashupStyle === 'feature-image-only' && (
          <ImagePrimary
            imageLayout={'intrinsic'}
            ratio="picture"
            fields={{
              primaryImageCaption: {
                value: '',
              },
              primaryImage:
                props.fields?.resultItems?.[0].templateId === ARTICLE_TEMPLATE_ID
                  ? featuredArticleDesktopImage
                  : featuredPrimaryImage,
              primaryImageMobile:
                props.fields?.resultItems?.[0].templateId === ARTICLE_TEMPLATE_ID
                  ? featuredArticleMobileImage
                  : featuredPrimaryImageMobile,
              primaryImageMobileFocusArea:
                props.fields?.resultItems?.[0].templateId === ARTICLE_TEMPLATE_ID
                  ? {
                      id: 'e6b36945-ebc0-4af8-ad95-d8268b56b7d0',
                      url: 'http://localhost/sitecore/login/sitecore/system/Settings/Feature/EnterpriseWeb/Enums/Image-Focus/Center',
                      name: 'Center',
                      displayName: 'Center',
                      fields: {
                        Value: {
                          value: 'center',
                        },
                      },
                      templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                      templateName: 'Enum',
                    }
                  : props.fields.resultItems[0].fields.primaryImageMobileFocusArea,
            }}
          />
        )}

        {props.fields?.displayEyebrow.value && (
          <Eyebrow
            useTag="span"
            fields={{
              eyebrowText: {
                value:
                  props.fields?.resultItems?.[0].templateId === ARTICLE_TEMPLATE_ID
                    ? getEnum<string>(props.fields?.resultItems?.[0].fields?.articleTopic) || ''
                    : props.fields?.resultItems?.[0].fields.eyebrow.value,
              },
            }}
            classes={classNames(
              props.mashupStyle === 'feature-image-only'
                ? themeData.classes.featuredCard.eyebrow.featuredImage
                : themeData.classes.featuredCard.eyebrow.noImages,
              'mt-s'
            )}
          />
        )}
        <Headline
          classes={classNames(
            props.mashupStyle === 'feature-image-only'
              ? themeData.classes.featuredCard.headline.featuredImage
              : themeData.classes.featuredCard.headline.noImages
          )}
          useTag="h3"
          fields={{
            headlineText: { value: props.fields?.resultItems?.[0].fields?.openGraphTitle?.value },
            superscriptCTA: {
              value: {
                href: '',
              },
            },
          }}
        />

        <RichTextWrapper
          classes={themeData.classes.featuredCard.body.featuredImage}
          field={{
            value: props.fields?.resultItems?.[0].fields.openGraphDescription.value,
          }}
        />

        <SingleButton
          classes={{ wrapper: 'mb-0 md:mb-s', cta1Classes: 'font-bold' }}
          fields={{
            cta1Link: {
              value: {
                href: props.fields?.resultItems?.[0].url,
                text: 'Read More',
                anchor: '',
                linktype: 'internal',
                class: '',
                title: '',
                target: '',
                querystring: '',
                id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
              },
            },
            cta1ModalLinkText: {
              value: '',
            },
            cta1Style: {
              id: '49a23327-0397-4cce-a930-e76918d37c42',
              url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
              name: 'Primary',
              displayName: 'Primary',
              fields: {
                Value: {
                  value: 'link',
                },
              },
              templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
              templateName: 'Enum',
            },
            cta1Icon: {
              id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
              url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
              name: 'Arrow',
              displayName: 'Arrow',
              fields: {
                Value: {
                  value: 'arrow',
                },
              },
              templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
              templateName: 'Enum',
            },
          }}
        />
      </div>

      <div className="md:gap-lg col-span-12  grid grid-cols-12 gap-x-s gap-y-ml self-start border-t border-gray pt-s md:col-span-6">
        {props.fields?.resultItems.slice(1)?.map((resultItem: ResultItem, index: number) => {
          return (
            <div key={index} className=" col-span-12  border-b border-gray ">
              {props.fields?.displayEyebrow.value && (
                <Eyebrow
                  useTag="span"
                  fields={{
                    eyebrowText: {
                      value:
                        resultItem?.templateId === ARTICLE_TEMPLATE_ID
                          ? getEnum<string>(resultItem.fields.articleTopic) || ''
                          : (resultItem?.fields.eyebrow as Field<string>).value,
                    },
                  }}
                  classes={themeData.classes.generalCard.eyebrow}
                />
              )}
              <Headline
                classes={themeData.classes.generalCard.headline}
                useTag="h3"
                fields={{
                  headlineText: {
                    value: (resultItem?.fields?.openGraphTitle as Field<string>)?.value,
                  },
                  superscriptCTA: {
                    value: {
                      href: '',
                    },
                  },
                }}
              />

              <RichTextWrapper
                classes={themeData.classes.generalCard.body}
                field={{
                  value: (resultItem?.fields.openGraphDescription as Field<string>).value,
                }}
              />
              <SingleButton
                classes={themeData.classes.generalCard.singleButton}
                fields={{
                  cta1Link: {
                    value: {
                      href: resultItem?.url,
                      text: 'Read More',
                      anchor: '',
                      linktype: 'internal',
                      class: '',
                      title: '',
                      target: '',
                      querystring: '',
                      id: '{7FB335D2-8E99-458E-9EF9-562A78CCB821}',
                    },
                  },
                  cta1ModalLinkText: {
                    value: '',
                  },
                  cta1Style: {
                    id: '49a23327-0397-4cce-a930-e76918d37c42',
                    url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/CTA-Styles/Primary',
                    name: 'Primary',
                    displayName: 'Primary',
                    fields: {
                      Value: {
                        value: 'link',
                      },
                    },
                    templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                    templateName: 'Enum',
                  },
                  cta1Icon: {
                    id: '50590edc-7ea7-4436-9a3e-701c87a07db2',
                    url: 'http://localhost/sitecore/login/sitecore/system/Settings/Foundation/EnterpriseWeb/Enums/Icons/Arrow',
                    name: 'Arrow',
                    displayName: 'Arrow',
                    fields: {
                      Value: {
                        value: 'arrow',
                      },
                    },
                    templateId: 'd2923fee-da4e-49be-830c-e27764dfa269',
                    templateName: 'Enum',
                  },
                }}
              />
            </div>
          );
        })}
      </div>
    </>
  );
};

export default FeaturedImageOnly;
