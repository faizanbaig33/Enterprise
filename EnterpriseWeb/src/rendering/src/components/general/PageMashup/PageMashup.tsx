// Global
import { Feature } from 'src/.generated/Feature.EnterpriseWeb.model';
import { useTheme } from 'lib/context/ThemeContext';
import { withDatasourceCheck } from '@sitecore-jss/sitecore-jss-nextjs';
// Components
import { Component } from 'src/helpers/Component';
import { PageMashupTheme } from './PageMashup.theme';
import { Headline } from 'src/helpers/Headline';
import { BodyCopy } from 'src/helpers/BodyCopy';
import SingleButton from 'src/helpers/SingleButton/SingleButton';
import { getEnum } from 'lib/utils';
import ImagesForAll from './ImagesForAll';
import FeaturedImageOnly from './FeaturedImageOnly';
import { ComponentProps } from 'lib/component-props';

export type MashupStyle = 'images-for-all' | 'feature-image-only' | 'no-images';

export type PageMashupProps = Feature.EnterpriseWeb.Components.General.PageMashup.PageMashup &
  ComponentProps;

const PageMashup = (props: PageMashupProps) => {
  const { themeData } = useTheme(PageMashupTheme);
  const MashupStyle = getEnum<MashupStyle>(props.fields?.mashupStyle) ?? 'images-for-all';

  const displayMashupGrid = () => {
    switch (MashupStyle) {
      case 'images-for-all':
        return <ImagesForAll {...props} />;
      case 'feature-image-only':
      case 'no-images':
        return <FeaturedImageOnly {...props} mashupStyle={MashupStyle} />;
      default:
        return <ImagesForAll {...props} />;
    }
  };

  return (
    <Component
      variant="full"
      padding="px-0"
      backgroundVariant={getEnum(props.fields?.backgroundColor)}
      dataComponent="general/pagemashup"
      {...props}
    >
      <div className="col-span-12 py-l">
        <div className="grid-rows-auto grid grid-cols-12  gap-s px-m  md:max-w-screen-lg lg:mx-auto">
          <div className="col-span-12 md:col-span-6">
            <Headline {...props} classes={themeData.classes.sectionheadline} />
          </div>
          <div className="col-span-12 md:col-span-6">
            <BodyCopy {...props} classes={themeData.classes.sectionBody} />
            <SingleButton {...props} classes={themeData.classes.sectionCta} />
          </div>

          {displayMashupGrid()}
        </div>
      </div>
    </Component>
  );
};

export default withDatasourceCheck()<PageMashupProps>(PageMashup);
