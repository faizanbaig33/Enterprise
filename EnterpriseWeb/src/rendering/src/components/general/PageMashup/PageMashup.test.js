// Global
import { hasDataComponent, snapshot } from 'lib/jest/test-utils';
// Local
import PageMashup from './PageMashup';
import imagesForAllData from './PageMashup.mock-data-images-for-all';

it('renders correctly', () => {
  const component = snapshot(PageMashup, { componentProps: imagesForAllData });
  hasDataComponent(component, 'general/pagemashup');
});
