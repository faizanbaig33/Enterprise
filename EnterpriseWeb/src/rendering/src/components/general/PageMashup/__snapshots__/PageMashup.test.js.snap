// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  data-component="general/pagemashup"
  id="id1751362294"
>
  <div
    class="my-8 w-full theme-white"
  >
    <div
      class="section-grid grid grid-cols-2 md:grid-cols-12 px-0 gap-s md:gap-s "
    >
      <div
        class="col-span-12 py-l"
      >
        <div
          class="grid-rows-auto grid grid-cols-12 gap-s px-m md:max-w-screen-lg lg:mx-auto"
        >
          <div
            class="col-span-12 md:col-span-6"
          >
            <div
              class="items-top font-sans  text-theme-text text-sm-m md:text-m font-heavy"
            >
              <h2>
                This is a featured headline lorem ipsum
              </h2>
              
            </div>
          </div>
          <div
            class="col-span-12 md:col-span-6"
          >
            <div
              class="text-theme-body text-body mb-m font-regular"
            >
              <div
                class="body-copy"
              >
                <div>
                  Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. Body text goes here. 
                </div>
              </div>
            </div>
            <div
              class="mb-s mb-s flex items-start md:flex-row md:items-center"
            >
              <a
                class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray font-heavy"
                href="/"
                role="link"
                tabindex="0"
                target=""
                title=""
              >
                Default State
                <span
                  class="ml-xxs"
                />
              </a>
            </div>
          </div>
          <div
            class="col-span-12 grid grid-cols-12 gap-s md:gap-s"
          >
            <div
              class="order-last col-span-full flex items-center  md:order-none md:col-span-4 "
            >
              <div>
                <div
                  class="font-heavy text-theme-text text-xxs mb-xxs md:mb-s"
                >
                  <span>
                    blog
                  </span>
                </div>
                <div
                  class="items-top font-sans  text-theme-text text-sm-m md:text-m font-heavy mb-s"
                >
                  <h3>
                    Headline lorem ipsum dolor amit lorem ipsum 1
                  </h3>
                  
                </div>
                <div
                  class="body-copy text-theme-body text-body mb-m md:mb-l font-regular"
                >
                  <div>
                    Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque. Delectus et veritatis voluptas quia ut. Mollitia dignissimos doloribus odit sit et est. Beatae quae sit sit consectetur.
                  </div>
                </div>
                <div
                  class="md:mb-0 mb-s flex items-start md:flex-row md:items-center"
                >
                  <a
                    class="flex w-fit items-center whitespace-nowrap rounded-lg border-4 border-theme-btn-border bg-theme-btn-bg px-m py-[9px] font-sans text-button font-heavy text-theme-btn-text hover:border-theme-btn-border-hover hover:bg-theme-btn-bg-hover hover:text-theme-btn-text-hover disabled:border-gray disabled:text-gray font-heavy"
                    href="/Article-Page-Mashup-1"
                    role="link"
                    tabindex="0"
                    target=""
                    title=""
                  >
                    Read More
                    <span
                      class="ml-xxs"
                    />
                  </a>
                </div>
              </div>
            </div>
            <div
              class="col-span-full md:col-span-8"
            >
              <div
                class="relative hidden md:block  "
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27998%27%20height=%27561%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="elf"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </div>
              <div
                class="relative block w-full md:hidden"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27998%27%20height=%27561%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="elf"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                  />
                  <noscript />
                </span>
              </div>
            </div>
          </div>
          <div
            class="col-span-12 grid grid-cols-2 gap-s border-b border-gray md:col-span-4 md:grid-cols-1 md:gap-s "
          >
            <div
              class="relative hidden md:block  "
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27998%27%20height=%27561%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="elf"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27998%27%20height=%27561%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="elf"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div>
              <div
                class="font-heavy text-theme-text text-xxs mb-xxs"
              >
                <span>
                  blog
                </span>
              </div>
              <div
                class="items-top font-sans  text-theme-text text-base font-serif font-heavy mb-xs"
              >
                <h3>
                  This is a card heading 2
                </h3>
                
              </div>
              <div
                class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
              >
                <div>
                  Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque.
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray font-heavy"
                  href="/Article-Page-Mashup-2"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="col-span-12 grid grid-cols-2 gap-s border-b border-gray md:col-span-4 md:grid-cols-1 md:gap-s "
          >
            <div
              class="relative hidden md:block  "
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271600%27%20height=%27900%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Img"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27392%27%20height=%27392%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="andersen big doors looking from the inside out to patio garden area"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div>
              <div
                class="font-heavy text-theme-text text-xxs mb-xxs"
              >
                <span>
                  Generic eyebrow
                </span>
              </div>
              <div
                class="items-top font-sans  text-theme-text text-base font-serif font-heavy mb-xs"
              >
                <h3>
                  This is a card headlin 3
                </h3>
                
              </div>
              <div
                class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
              >
                <div>
                  Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque.
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray font-heavy"
                  href="/Generic-Page-Mashup-1"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
          <div
            class="col-span-12 grid grid-cols-2 gap-s border-b border-gray md:col-span-4 md:grid-cols-1 md:gap-s "
          >
            <div
              class="relative hidden md:block  "
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%271600%27%20height=%27900%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="Img"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
                <noscript />
              </span>
            </div>
            <div
              class="relative block w-full md:hidden"
            >
              <span
                style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
              >
                <span
                  style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27392%27%20height=%27392%27/%3e"
                    style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                  />
                </span>
                <img
                  alt="andersen big doors looking from the inside out to patio garden area"
                  data-nimg="intrinsic"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover; object-position: center;"
                />
                <noscript />
              </span>
            </div>
            <div>
              <div
                class="font-heavy text-theme-text text-xxs mb-xxs"
              >
                <span>
                  Generic eyebrow
                </span>
              </div>
              <div
                class="items-top font-sans  text-theme-text text-base font-serif font-heavy mb-xs"
              >
                <h3>
                  This is a card headline 4
                </h3>
                
              </div>
              <div
                class="body-copy text-theme-body text-body mb-xxs md:mb-s font-regular line-clamp-3 md:line-clamp-none"
              >
                <div>
                  Est voluptatum tempore nihil provident alias quis enim. Possimus voluptatibus deserunt. Ea sint repellendus nihil eligendi neque.
                </div>
              </div>
              <div
                class="mb-s flex items-start md:flex-row md:items-center"
              >
                <a
                  class="flex w-fit items-center whitespace-nowrap font-sans text-text-link font-heavy text-theme-text hover:underline hover:decoration-secondary hover:underline-offset-8 disabled:border-gray disabled:text-gray font-heavy"
                  href="/Generic-Page-Mashup-1"
                  role="link"
                  tabindex="0"
                  target=""
                  title=""
                >
                  Read More
                  <span
                    class="ml-xxs"
                  />
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
